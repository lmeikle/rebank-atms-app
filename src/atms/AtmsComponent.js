import React from 'react';
import { string } from 'prop-types';
import './Atms.css';

/**
 * Stateless component which renders a single ATM component.
 * Clicking on it will launch google maps using the ATM's coordinates
 */
const AtmsComponent = ({ name, countryCode, address, coords, distance, numOfAtms }) => {
  const googleMapsUrl = `http://www.google.com/maps/place/${coords.Latitude},${coords.Longitude}`;
  return (
    <a href={googleMapsUrl} target="_blank">
      <div className="AtmContainer">
        <div>Bank Name: {name}</div>
        <div>Address: {address}</div>
        <div>Country: {countryCode}</div>
        <div>Distance: {distance.toFixed(2)}km</div>
        <div>Number of ATM's: {numOfAtms}</div>
      </div>
    </a>
  );
};

AtmsComponent.propTypes = {
  name: string.isRequired,
  countryCode: string.isRequired,
  address: string.isRequired
};

AtmsComponent.defaultProps = {};

export default AtmsComponent;
