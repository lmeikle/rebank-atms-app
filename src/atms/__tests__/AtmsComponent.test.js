import React from 'react';
import { shallow } from 'enzyme';
import AtmsComponent from '../AtmsComponent';

let props = {
  name: 'Barclays Bank',
  countryCode: 'UK',
  address: '10 Camden High Street',
  coords: { Latitude: 123, Longitude: 456 },
  distance: 1.23,
  numOfAtms: 3
};

const atmsComponent = shallow(<AtmsComponent {...props} />);

test('renders correctly', () => {
  expect(atmsComponent).toMatchSnapshot();
});
