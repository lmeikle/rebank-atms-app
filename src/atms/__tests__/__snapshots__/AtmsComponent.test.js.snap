// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AtmsComponent
    address="10 Camden High Street"
    coords={
        Object {
            "Latitude": 123,
            "Longitude": 456,
          }
    }
    countryCode="UK"
    distance={1.23}
    name="Barclays Bank"
    numOfAtms={3}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="atm"
>
        <div>
                Bank Name: 
                Barclays Bank
        </div>
        <div>
                Address: 
                10 Camden High Street
        </div>
        <div>
                Country: 
                UK
        </div>
        <div>
                Distance: 
                1.23
                km
        </div>
        <div>
                Number of ATM's: 
                3
        </div>
</div>,
      "href": "http://www.google.com/maps/place/123,456",
      "target": "_blank",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            Bank Name: 
            Barclays Bank
</div>,
          <div>
            Address: 
            10 Camden High Street
</div>,
          <div>
            Country: 
            UK
</div>,
          <div>
            Distance: 
            1.23
            km
</div>,
          <div>
            Number of ATM's: 
            3
</div>,
        ],
        "className": "atm",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Bank Name: ",
              "Barclays Bank",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Bank Name: ",
            "Barclays Bank",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Address: ",
              "10 Camden High Street",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Address: ",
            "10 Camden High Street",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Country: ",
              "UK",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Country: ",
            "UK",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Distance: ",
              "1.23",
              "km",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Distance: ",
            "1.23",
            "km",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Number of ATM's: ",
              3,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Number of ATM's: ",
            3,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "a",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="atm"
>
          <div>
                    Bank Name: 
                    Barclays Bank
          </div>
          <div>
                    Address: 
                    10 Camden High Street
          </div>
          <div>
                    Country: 
                    UK
          </div>
          <div>
                    Distance: 
                    1.23
                    km
          </div>
          <div>
                    Number of ATM's: 
                    3
          </div>
</div>,
        "href": "http://www.google.com/maps/place/123,456",
        "target": "_blank",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              Bank Name: 
              Barclays Bank
</div>,
            <div>
              Address: 
              10 Camden High Street
</div>,
            <div>
              Country: 
              UK
</div>,
            <div>
              Distance: 
              1.23
              km
</div>,
            <div>
              Number of ATM's: 
              3
</div>,
          ],
          "className": "atm",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Bank Name: ",
                "Barclays Bank",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Bank Name: ",
              "Barclays Bank",
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Address: ",
                "10 Camden High Street",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Address: ",
              "10 Camden High Street",
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Country: ",
                "UK",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Country: ",
              "UK",
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Distance: ",
                "1.23",
                "km",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Distance: ",
              "1.23",
              "km",
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Number of ATM's: ",
                3,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Number of ATM's: ",
              3,
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "a",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
